### ISSUE TITLE
Cannot assign to read only property

### ISSUE THREAD

**Issue by Prontsevich** [timdeschryver/ngrx-etc#14]
Hello!

I'm using your package in my project. And I got this error today:
`ERROR TypeError: Cannot assign to read only property 'isActive' of object '[object Object]'`
How can I fix it?

My reducer:
```
export interface DatasetInfoState {
  datasetInfo: DatasetPublisherInfo;
  callState: CallState;
}

export const initialState: DatasetInfoState = {
  datasetInfo: null,
  callState: LoadingState.INIT,
};

const datasetInfoReducer = createReducer(
  initialState,
  mutableOn(fromActions.DatasetInfoActions.publishDatasetSuccess, (draft: DatasetInfoState) => {
    draft.callState = LoadingState.LOADED;
    draft.datasetInfo.isActive = true;
  }),

  mutableOn(fromActions.DatasetInfoActions.unpublishDatasetSuccess, (draft: DatasetInfoState) => {
    draft.callState = LoadingState.LOADED;
    draft.datasetInfo.isActive = false;
  })
);

export function reducer(state: DatasetInfoState | undefined, action: Action) {
  return datasetInfoReducer(state, action);
}
```
Angular 9.1.3
Ngrx 9.0.0
Ngrx-etc 4.0.0

**Comment by timdeschryver** [comment]
Do you have a reproduction of this issue?
My guess would be that you're modifying the state outside reducers.

**Comment by Prontsevich** [comment]
I will try to create a demo to reproduce it. But according to error log the issue raised in reducer when I'm trying to change isActive property.
If I change code like that - everything works good.
```
const dataset = {...draft.datasetInfo};
dataset.isActive = false;
draft.datasetInfo = dataset;
```

**Comment by timdeschryver** [comment]
Thanks! Hmm that's weird

**Comment by Prontsevich** [comment]
I wasn't able to reproduce it in Stackblitz. But here is the screnshot of this error. Maybe you have some ideas. 
![image](https://user-images.githubusercontent.com/1219707/82051237-2cf07780-96c2-11ea-8c03-e3cefbb68e75.png)

**Comment by timdeschryver** [comment]
@Prontsevich I was also unable to reproduce it, see the [extra test](https://github.com/timdeschryver/ngrx-etc/commit/78e6b301d61201fef8a4ad70bb41be88a9e21e0a).

I'm going to close this issue for now if you can reproduce it, feel free to re-open the issue.

**Comment by kamilchlebek** [comment]
@timdeschryver I think I've found the issue. It happened to me and seems it was caused because` immerjs` was installed twice. Once inside `ngrx-etc` because of `dependency` instead of `peerDependency` and second directly inside `node_modules`. 

Seems that `ngrx-etc` should not have `immerjs` inside `dependencies` to avoid it. I couldn't avoid adding `immerjs` to my project `package.json` because I wanted to use `setAutoFreeze`

Also I noticed something weird. When I'm using `mutableOn` my state does not get refresh after editing typescript Class (state is updated, but change detection is not triggered). Somehow replacing `mutableOn` with this function (https://github.com/immerjs/immer/issues/252#issuecomment-504026822) solves the issue.
